Letterbook.Workers/Consumers/OutboundPostConsumerDefinition.cs

```diff
@@ -0,0 +1,28 @@
+using Letterbook.Workers.Publishers;
+using MassTransit;
+using MassTransit.Middleware;
+
+namespace Letterbook.Workers.Consumers;
+
+public class OutboundPostConsumerDefinition :
+	ConsumerDefinition<OutboundPostConsumer>
+{
+	protected override void ConfigureConsumer(IReceiveEndpointConfigurator endpoint,
+		IConsumerConfigurator<OutboundPostConsumer> consumer, IRegistrationContext context)
+	{
+		consumer.UseFilter(new ContextFilter<ConsumerConsumeContext<OutboundPostConsumer>>(async consumeContext =>
+		{
+			await Task.FromResult(0);
+			return consumeContext.Headers.TryGetHeader(Headers.Event, out var header) &&
+			       header switch
+			       {
+				       nameof(PostEventPublisher.Published) => true,
+				       nameof(PostEventPublisher.Updated) => true,
+				       nameof(PostEventPublisher.Deleted) => true,
+				       nameof(PostEventPublisher.Shared) => true,
+				       nameof(PostEventPublisher.Liked) => true,
+				       _ => false
```

Comment:
Consumer Definitions allow us to customize the message pipeline for a consumer class. This specifically is filtering for the event types that OutboundPostConsumer is meant to handle. This happens before the message body is deserialized; at this point only the headers should have gone through deserialization. So it is both more efficient and somewhat simplifies the consumer implementation.